### PR TITLE
Use actual passed argument to intrinsic functions as result not the casted argument

### LINKED ISSUE #5385: (PRIMA) Phisical type issue for transpose in `intrinsic_pass`
Towards #4397 
It is almost same as #5334 . Just Difference is only line I Comented below.

MRE : 
```fortran 
program arrays_57

real :: A(2,2) = reshape([1.0,2.0,3.0,4.0], [2,2])
call trans(A)
print *, A 

contains

subroutine trans(A)
real, intent(inout) :: A(:, :)
real :: B(size(A,1), size(A,2))  !! This line creates local variable with size same as A 
B = A
A = matprod(transpose(B)) !! Local variable is Passed in transpose
end subroutine

function matprod(x) result(k)
real,intent(in) :: x(:,:)
real :: k(size(x, 1), size(x, 2))
k = 12.91
end function

end program 
```

LFortran Error : 
```console
ASR verify pass error: Old physical type conflicts with the physical type of argument 1 0
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  Binary file "/mnt/c/lfortran/src/bin/lfortran", in _start
  File "/build/glibc-LcI20x/glibc-2.31/csu/../csu/libc-start.c", line 308, in __libc_start_main
  File "/mnt/c/lfortran/src/bin/lfortran.cpp", line 2686, in main
    return main_app(argc, argv);
  File "/mnt/c/lfortran/src/bin/lfortran.cpp", line 2628, in main_app(int, char**)
    err = compile_src_to_object_file(arg_file, tmp_o, false,
  File "/mnt/c/lfortran/src/bin/lfortran.cpp", line 1045, in (anonymous namespace)::compile_src_to_object_file(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, LCompilers::CompilerOptions&, LCompilers::PassManager&)
    res = fe.get_llvm3(*asr, lpm, diagnostics, infile);
  File "/mnt/c/lfortran/src/lfortran/fortran_evaluator.cpp", line 383, in LCompilers::FortranEvaluator::get_llvm3(LCompilers::ASR::TranslationUnit_t&, LCompilers::PassManager&, LCompilers::diag::Diagnostics&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    compiler_options, run_fn, infile);
  File "/mnt/c/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 10361, in LCompilers::asr_to_llvm(LCompilers::ASR::TranslationUnit_t&, LCompilers::diag::Diagnostics&, llvm::LLVMContext&, Allocator&, LCompilers::PassManager&, LCompilers::CompilerOptions&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    pass_manager.apply_passes(al, &asr, co.po, diagnostics);
  File "/mnt/c/lfortran/src/libasr/pass/pass_manager.h", line 401, in LCompilers::PassManager::apply_passes(Allocator&, LCompilers::ASR::TranslationUnit_t*, LCompilers::PassOptions&, LCompilers::diag::Diagnostics&)
    apply_passes(al, asr, _passes, pass_options, diagnostics);
LCompilersException: Verify failed in the pass: intrinsic_function
```

### PR DESCRIPTION
**Pull request by assem2002**
fixes #5385

### CODE REVIEW

**Review by certik — APPROVED** [review]
Fix https://github.com/lfortran/lfortran/pull/5422/files#r1850547750, then you can merge.

File: src/libasr/pass/intrinsic_function.cpp [hunk 1]
```diff
@@ -485,8 +485,26 @@ class ReplaceFunctionCallReturningArray: public ASR::BaseExprReplacer<ReplaceFun
...
                 al, x->base.base.loc, x->m_name, x->m_original_name, new_args.p,
                 new_args.size(), x->m_dt, nullptr, false, false)));
-
-            *current_expr = new_args.p[new_args.size() - 1].m_value;
+            ASR::expr_t* subroutineCall_res = new_args.p[new_args.size() - 1].m_value;
+            // Avoid ArrayPhysicalCasting. Use actual argument.
+            // intrinsicArrayFunction has a return that's handled be the super expression. Don't use casted expr created by `make_SubroutineCall_t_uti()`. 
+            while(ASR::is_a<ASR::ArrayPhysicalCast_t>(*subroutineCall_res)){
+                subroutineCall_res = ASRUtils::get_past_array_physical_cast(subroutineCall_res);
+            }
```
[the remark this answers is not in the document]
Comment (Pranavchiku) [reply src/libasr/pass/intrinsic_function.cpp:493]: I had a look at this issue and was trying to understand why we have nested ArrayPhysicalCast in first place and it comes through "recursive" behaviour in array_op which took considerable amount of time to understand but the functioning was correct so I couldn't fix it there. Hence, this solution looked good to me.

[the remark this answers is not in the document]
Comment (certik) [reply src/libasr/pass/intrinsic_function.cpp:493]: Ok. Good for now, we are reworking array_op anyway, once we switch to the simplifier, we can fix this casting there. Until then this should be ok.

[the remark this answers is not in the document]
Comment (assem2002) [reply src/libasr/pass/intrinsic_function.cpp:493]: I like the idea of dissolving nested `ArrayPhysicalCast` in some pass. but let's wait for the simplifier to be completely working; for instance it handles the test case provided in this PR without needing the changes introduced here.

**Pull request merged**